### PR TITLE
test(grid-pinning): Add tests for navigation to/from pinned area #6676

### PR DESCRIPTION
**Pull request by dkamburov**
Closes #  

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid/grid.pinning.spec.ts [hunk 1]
```diff
@@ -417,6 +418,41 @@ describe('IgxGrid - Column Pinning #grid', () => {
...
+                rowEnd: cellContactName.rowIndex,
+                columnStart: cellContactName.visibleColumnIndex,
+                columnEnd: cellContactName.visibleColumnIndex
+            };
+            grid.selectRange(range);
+            grid.navigation.activeNode = {row: cellContactName.rowIndex, column: cellContactName.visibleColumnIndex};
+            fix.detectChanges();
+
+            const keydownArrowRightEvent = new KeyboardEvent('keydown', {key: 'ArrowRight'});
+            grid.navigation.dispatchEvent(keydownArrowRightEvent);
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.pinning.spec.ts]: You can use:
grid.navigation.dispatchEvent(UIInteractions.getKeyboardEvent('keydown', 'ArrowDown'))

File: projects/igniteui-angular/src/lib/grids/grid/grid.pinning.spec.ts [hunk 2]
```diff
@@ -639,6 +675,65 @@ describe('IgxGrid - Column Pinning #grid', () => {
...
+
+            const cellCompanyName = grid.getCellByColumn(0, 'CompanyName');
+            const range = { rowStart: 0, rowEnd: 0, columnStart: 9, columnEnd: 9 };
+            grid.selectRange(range);
+            grid.navigation.activeNode = {row: 0, column: 9};
+            fix.detectChanges();
+            expect(cellCompanyName.active).toBe(true);
+
+            const keydownArrowLeftEvent = new KeyboardEvent('keydown', {key: 'ArrowLeft', ctrlKey: true});
+            grid.navigation.dispatchEvent(keydownArrowLeftEvent);
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.pinning.spec.ts]: Same as the other you can use:
 grid.navigation.dispatchEvent(UIInteractions.getKeyboardEvent('keydown', 'ArrowDown', false, false, true))

**Comment by kdinev** [comment]
@dkamburov Can you please look at the build errors?

**Comment by ChronosSF** [comment]
> @dkamburov Can you please look at the build errors?

This is based on @ddincheva 's navigation changes so it'll only work once it is in master. This was done to prevent having to rewrite the tests shortly after committing them

**Pull request merged**